src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,37 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add a country to DB: " + country);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return country;
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Country country;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            country = session.get(Country.class, id);
+            session.close();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get country by id from DB: " + id);
+        }
+        return Optional.ofNullable(country);
```

Comment:
Use try with finally, where you use transaction and try with resources with Read operations.
see https://mate-academy.github.io/jv-program-common-mistakes/hibernate/relations/relations_checklist